### PR TITLE
Adding Identifier Uris as an input option to be used for Saml config if enabled.

### PR DESCRIPTION
**Pull request by MartyFox**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-15781

### Change description ###
Adding Identifier Uris as an input option to be used for Saml config if enabled.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -133,7 +135,21 @@ for await (const app of apps) {
...
+      app.preferredSingleSignOnMode &&
+      app.preferredSingleSignOnMode == "saml"
+    ) {
+      await addSsoSamlUris({
+        token: token,
+        appId: applicationId,
+        identifierUrls: app.identifierUrls,
+        redirectUrls: app.redirectUrls,
+      });
+    }
```
Comment (timja) [inline src/main.ts:149]: why do we need this? it's already here?
https://github.com/hmcts/azure-app-proxy-manager/blob/59649c213b24b7441827245b6d1a79214209343d/src/applicationManager.ts#L236-L238

Comment (MartyFox) [reply src/main.ts:149]: The `updateApplicationConfig` function doesnt appear to deploy the identifier Urls first time, I had to re-run for those and the extra redirect Urls to be added, very possible I'm missing something though.

File: src/main.ts [hunk 2]
```diff
@@ -133,6 +135,19 @@ for await (const app of apps) {
         appRoles: app.appRoles,
       });
     }
+
+    if (
+      app.preferredSingleSignOnMode &&
+      app.preferredSingleSignOnMode == "saml"
+    ) {
+      await addSsoSamlUris({
```
Comment (timja) [inline src/main.ts:143]: are these specific to saml?

https://learn.microsoft.com/en-us/entra/identity-platform/reference-app-manifest#identifieruris-attribute

seems to be general purpose, would it be better to not gate behind SAML and instead just do it based on identifier urls being set?

**Review by timja — APPROVED** [review]

File: apps.schema.json [hunk 3]
```diff
@@ -207,6 +207,18 @@
             },
             "examples": [["https://my-external-facing-url.domain.com/redirect"]]
           },
+          "identifierUrls": {
+            "type": "array",
+            "default": [],
+            "title": "A list of identifier Urls that will be added to the application SAML config when preferredSingleSignOnMode set to `saml`. Defaults to externalUrl if not set",
```
Comment (timja) [inline apps.schema.json:213]: this description needs updating and the items title

**Review by timja — APPROVED** [review]

**Pull request merged**